taf/testlib/tempest_clients/magnum/clients/cluster_client.py

```diff
@@ -0,0 +1,175 @@
+        """
+
+        return "{0}/{1}".format(cls.clusters_uri(), cluster_id)
+
+    def list_clusters(self, filters=None, **kwargs):
+        """Makes GET /clusters request and returns ClusterCollection
+
+        Abstracts REST call to return all clusters
+
+        :param filters: Optional k:v dict that's converted to url query
+        :returns: response object and ClusterCollection object
+        """
+
+        resp, body = self.get(self.clusters_uri(filters), **kwargs)
+        return self.deserialize(resp, body, cluster_model.ClusterCollection)
+
+    def get_cluster(self, cluster_id, **kwargs):
+        """Makes GET /cluster request and returns ClusterEntity
+
+        Abstracts REST call to return a single cluster based on uuid or name
+
+        :param cluster_id: cluster uuid or name
+        :returns: response object and ClusterCollection object
+        """
+
+        resp, body = self.get(self.cluster_uri(cluster_id))
+        return self.deserialize(resp, body, cluster_model.ClusterEntity)
+
+    def post_cluster(self, model, **kwargs):
+        """Makes POST /cluster request and returns ClusterIdEntity
+
+        Abstracts REST call to create new cluster
+
+        :param model: ClusterEntity
+        :returns: response object and ClusterIdEntity object
+        """
+
+        resp, body = self.post(
+            self.clusters_uri(),
+            body=model.to_json(), **kwargs)
+        return self.deserialize(resp, body, cluster_id_model.ClusterIdEntity)
+
+    def patch_cluster(self, cluster_id, clusterpatch_listmodel, **kwargs):
+        """Makes PATCH /cluster request and returns ClusterIdEntity
+
+        Abstracts REST call to update cluster attributes
+
+        :param cluster_id: UUID of cluster
+        :param clusterpatch_listmodel: ClusterPatchCollection
+        :returns: response object and ClusterIdEntity object
+        """
+
+        resp, body = self.patch(
+            self.cluster_uri(cluster_id),
+            body=clusterpatch_listmodel.to_json(), **kwargs)
+        return self.deserialize(resp, body, cluster_id_model.ClusterIdEntity)
+
+    def delete_cluster(self, cluster_id, **kwargs):
+        """Makes DELETE /cluster request and returns response object
+
+        Abstracts REST call to delete cluster based on uuid or name
+
+        :param cluster_id: UUID or name of cluster
+        :returns: response object
+        """
+
+        return self.delete(self.cluster_uri(cluster_id), **kwargs)
+
+    def wait_for_cluster_to_delete(self, cluster_id):
+        wait_for_condition(
+            lambda: self.does_cluster_not_exist(cluster_id), 10, 600)
+
+    def wait_for_created_cluster(self, cluster_id, delete_on_error=True):
+        try:
+            wait_for_condition(
+                lambda: self.does_cluster_exist(cluster_id), 10, 1800)
+        except Exception:
+            # In error state.  Clean up the cluster id if desired
+            self.LOG.error('Cluster %s entered an exception state.' %
```

Comment:
This magnum client code is taken over from magnum project. I wanted to keep it in the original shape. Except for some imports adjustments, I haven't touched it.